analytics/src/main/java/com/segment/analytics/Analytics.java

```diff
@@ -55,6 +58,7 @@ public class Analytics {
     this.messageTransformers = messageTransformers;
     this.messageInterceptors = messageInterceptors;
     this.log = log;
+    this.externalContext = null;
```

Comment:
context needs to be added in a MessageTransformer as it needs to be added when the message is being built.  If we are in a server side app dropping segment tracking events, it would need to have picked up the context for that individual request which is traditionally ThreadLocalContext. Its not on the Analytics object itself.